pkg/util/azureenvironment/environment.go

```diff
@@ -0,0 +1,24 @@
+package azureenvironment
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"os"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// AzureEnvFromEnvVar returns pointer to Azure cloud environment corresponding to AZURE_ENVIRONMENT.
+// Defaults to Public Cloud.
+func AzureEnvFromEnvVar() (*azure.Environment, error) {
```

Comment:
if we use env.Core everywhere, hopefully this goes away again?

Reply:
it's gone